http-client/src/main/java/com/facebook/airlift/http/client/jetty/ConcurrentScheduler.java

```diff
@@ -1,61 +1,57 @@
 package com.facebook.airlift.http.client.jetty;
 
+import com.facebook.airlift.concurrent.ConcurrentScheduledExecutor;
 import org.eclipse.jetty.util.component.AbstractLifeCycle;
 import org.eclipse.jetty.util.thread.Scheduler;
 
+import java.util.Optional;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.ScheduledFuture;
-import java.util.concurrent.ScheduledThreadPoolExecutor;
-import java.util.concurrent.ThreadFactory;
-import java.util.concurrent.ThreadLocalRandom;
 import java.util.concurrent.TimeUnit;
 
-import static com.facebook.airlift.concurrent.Threads.daemonThreadsNamed;
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
-// based on ScheduledExecutorScheduler
-class ConcurrentScheduler
+public class ConcurrentScheduler
         extends AbstractLifeCycle
         implements Scheduler
 {
+    private Optional<ConcurrentScheduledExecutor> concurrentScheduler;
+    private final int schedulerCount;
     private final int threadsPerScheduler;
```

Comment:
Does threadsPerScheduler make sense now that we can have as many single threaded schedulers as we need? Should that configuration be removed?

Reply:
I still have this question. Does having multiple threads per scheduler still make sense?

Reply:
Why each scheduler should have only one thread,   I didn't change the behavior of this class. See example ``http-client.timeout-threads`` and ``http-client.timeout-concurrency``